tests/jballerina-unit-test/src/test/resources/test-src/types/typedesc/typedesc_negative_runtime.bal

```diff
@@ -0,0 +1,126 @@
+const AAA = "AAA";
+
+type RecordType record {|
+    string name;
+|};
+
+type JsonTypeDesc typedesc<json>;
+type MapTypeDesc typedesc<map<string>>;
+type TableTypeDesc typedesc<table<map<string>>>;
+type MapJsonTypeDesc typedesc<map<JsonTypeDesc>>;
+type StringTypeDesc typedesc<string>;
+type IntTypeDesc typedesc<int>;
+type BooleanTypeDesc typedesc<boolean>;
+type FloatTypeDesc typedesc<float>;
+type DecimalTypeDesc typedesc<decimal>;
+type XMlTypeDesc typedesc<xml>;
+type BytesTypeDesc typedesc<byte>;
+type AnyTypeDesc typedesc<any>;
+type NullTypeDesc typedesc<null>;
+type AnydataTypeDesc typedesc<anydata>;
+type ObjectTypeDesc typedesc<object{}>;
+type ArrayTypeDesc typedesc<anydata[]>;
+type UnionTypeDesc typedesc<json|map<string>>;
+type IntersectionTypeDesc typedesc<int & readonly>;
+type TupleTypeDesc typedesc<[int, string, decimal]>;
+type SingletonTypeDesc typedesc<AAA>;
+
+function jsonTypeDesc() returns error? {
+    RecordType value = {name: "test"};
+    var _ = check value.ensureType(JsonTypeDesc);
```

Comment:
Adding tests with `ensureType` will check this issue with `ensureType` function.
there can be some other function that may affect with this change

Reply:
In the original [#issue](https://github.com/ballerina-platform/ballerina-lang/issues/36543#issuecomment-1511843600) code crashed happens due to a CC exception when we are passing `typedesc` to native functions. So changing the functions doesn't have any impact. Anyway I will add few more tests few more functions.

Reply:
@pcnfernando wdyt?